Formula/l/locust.rb

```diff
@@ -23,6 +23,7 @@ class Locust < Formula
   depends_on "python-typing-extensions"
   depends_on "python@3.12"
   depends_on "six"
+  depends_on "zeromq"
```

Comment:
All the other formulae using pyzmq (condure, jupyterlab, salt, zurl) depend on zeromq. If the dependency is not included, then pyzmq will use it's bundled zeromq.

https://github.com/Homebrew/homebrew-core/actions/runs/7004726049/job/19053226567#step:3:3113

Reply:
make sense, thanks!

Reply:
copy out the error log for future ref

```
==> python3.12 -m pip --python=/opt/homebrew/Cellar/locust/2.19.1/libexec/bin/python install --verbose --no-deps --no-binary=:all: --ignore-installed --no-compile /private/tmp/locust--pyzmq-20231127-6626-4putkf/pyzmq-25.1.1
  Using pip 23.2.1 from /opt/homebrew/lib/python3.12/site-packages/pip (python 3.12)
  Processing /private/tmp/locust--pyzmq-20231127-6626-4putkf/pyzmq-25.1.1
    Installing build dependencies: started
    Running command pip subprocess to install build dependencies
    Ignoring cffi: markers 'implementation_name == "pypy"' don't match your environment
    Collecting setuptools
      Using cached setuptools-69.0.2-py3-none-any.whl
    Collecting setuptools_scm[toml]
      Downloading setuptools-scm-8.0.4.tar.gz (74 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.3/74.3 kB 3.7 MB/s eta 0:00:00
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Installing backend dependencies: started
      Installing backend dependencies: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
    Collecting wheel
      Using cached wheel-0.42.0-py3-none-any.whl
    Collecting packaging
      Downloading packaging-23.2.tar.gz (146 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 146.7/146.7 kB 3.1 MB/s eta 0:00:00
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
    Collecting cython>=0.29
      Using cached Cython-3.0.6-cp312-cp312-macosx_14_0_arm64.whl
    Collecting typing-extensions (from setuptools_scm[toml])
      Downloading typing_extensions-4.8.0.tar.gz (71 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 71.5/71.5 kB 2.9 MB/s eta 0:00:00
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
    Building wheels for collected packages: packaging, setuptools_scm, typing-extensions
      Building wheel for packaging (pyproject.toml): started
      Building wheel for packaging (pyproject.toml): finished with status 'done'
      Created wheel for packaging: filename=packaging-23.2-py3-none-any.whl size=5[3011](https://github.com/Homebrew/homebrew-core/actions/runs/7004726049/job/19053226567#step:3:3012) sha256=ff253a4d6baae891d17264c4d3db39e5c545530a0790445dc9d81e17d8d3642a
      Stored in directory: /Users/brew/Library/Caches/Homebrew/pip_cache/wheels/21/3a/c1/6fdff902fd8cbfc8240a6fd5b598f653ad0be3809997ed6c4c
      Building wheel for setuptools_scm (pyproject.toml): started
      Building wheel for setuptools_scm (pyproject.toml): finished with status 'done'
      Created wheel for setuptools_scm: filename=setuptools_scm-8.0.4-py3-none-any.whl size=42139 sha256=f95b20f199747865f336a22ce110cb828533c4aba27ca38be9240e1eb444c7b0
      Stored in directory: /Users/brew/Library/Caches/Homebrew/pip_cache/wheels/a8/0d/c1/b048cc0c70c86a3b35ee6aa3a3368a12d9b2c4f9bba926eb63
      Building wheel for typing-extensions (pyproject.toml): started
      Building wheel for typing-extensions (pyproject.toml): finished with status 'done'
      Created wheel for typing-extensions: filename=typing_extensions-4.8.0-py3-none-any.whl size=31584 sha256=67ee37f832e3797940fb0b72813ba7d5e9f2bc0bbf007e6124534d1635128d66
      Stored in directory: /Users/brew/Library/Caches/Homebrew/pip_cache/wheels/1c/55/96/6fa06da5abda014f8f401c269435d693ef2aeba6c8fc1c9249
    Successfully built packaging setuptools_scm typing-extensions
    Installing collected packages: wheel, typing-extensions, setuptools, packaging, cython, setuptools_scm
    Successfully installed cython-3.0.6 packaging-23.2 setuptools-69.0.2 setuptools_scm-8.0.4 typing-extensions-4.8.0 wheel-0.42.0
  
  Notice: e] A new release of pip is available: 23.2.1 -> 23.3.1
  Notice: e] To update, run: python -m pip install --upgrade pip
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Running command Getting requirements to build wheel
    performance hint: zmq/backend/cython/message.pyx:78:5: Exception check on 'free_python_msg' will always require the GIL to be acquired.
    Possible solutions:
          1. Declare the function as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.
          2. Use an 'int' return type on the function to allow an error code to be returned.
    Compiling zmq/backend/cython/_device.pyx because it changed.
    Compiling zmq/backend/cython/_poll.pyx because it changed.
    Compiling zmq/backend/cython/_proxy_steerable.pyx because it changed.
    Compiling zmq/backend/cython/_version.pyx because it changed.
    Compiling zmq/backend/cython/context.pyx because it changed.
    Compiling zmq/backend/cython/error.pyx because it changed.
    Compiling zmq/backend/cython/message.pyx because it changed.
    Compiling zmq/backend/cython/socket.pyx because it changed.
    Compiling zmq/backend/cython/utils.pyx because it changed.
    Compiling zmq/devices/monitoredqueue.pyx because it changed.
    [ 1/10] Cythonizing zmq/backend/cython/_device.pyx
    [ 2/10] Cythonizing zmq/backend/cython/_poll.pyx
    [ 3/10] Cythonizing zmq/backend/cython/_proxy_steerable.pyx
    [ 4/10] Cythonizing zmq/backend/cython/_version.pyx
    [ 5/10] Cythonizing zmq/backend/cython/context.pyx
    [ 6/10] Cythonizing zmq/backend/cython/error.pyx
    [ 7/10] Cythonizing zmq/backend/cython/message.pyx
    [ 8/10] Cythonizing zmq/backend/cython/socket.pyx
    [ 9/10] Cythonizing zmq/backend/cython/utils.pyx
    [10/10] Cythonizing zmq/devices/monitoredqueue.pyx
    /private/tmp/pip-build-env-3kvk14o2/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py:265: UserWarning: Unknown distribution option: 'cffi_modules'
      warnings.warn(msg)
    running egg_info
    writing pyzmq.egg-info/PKG-INFO
    writing dependency_links to pyzmq.egg-info/dependency_links.txt
    writing requirements to pyzmq.egg-info/requires.txt
    writing top-level names to pyzmq.egg-info/top_level.txt
    running configure
    /opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config: line 12: /opt/homebrew/opt/pkg-config/bin/pkg-config: No such file or directory
    /opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config: line 12: exec: /opt/homebrew/opt/pkg-config/bin/pkg-config: cannot execute: No such file or directory
    Did not find libzmq via pkg-config.
    {'libraries': ['zmq'], 'include_dirs': [], 'library_dirs': [], 'runtime_library_dirs': [], 'extra_link_args': []}
    clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -I/opt/homebrew/Cellar/locust/2.19.1/libexec/include -I/opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/include/python3.12 -c build/temp.macosx-14-arm64-cpython-312/scratch/check_sys_un.c -o build/temp.macosx-14-arm64-cpython-312/scratch/check_sys_un.o
    clang -undefined dynamic_lookup build/temp.macosx-14-arm64-cpython-312/scratch/check_sys_un.o -o build/temp.macosx-14-arm64-cpython-312/scratch/check_sys_un
    Configure: Autodetecting ZMQ settings...
        Custom ZMQ dir:
    clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -Izmq/utils -c build/temp.macosx-14-arm64-cpython-312/scratch/vers.c -o build/temp.macosx-14-arm64-cpython-312/scratch/vers.o
    build/temp.macosx-14-arm64-cpython-312/scratch/vers.c:4:10: fatal error: 'zmq.h' file not found
    #include "zmq.h"
             ^~~~~~~
    1 error generated.
  
    error: command '/opt/homebrew/Library/Homebrew/shims/mac/super/clang' failed with exit code 1
  
    Failed with default libzmq, trying again with /usr/local
    {'libraries': ['zmq'], 'include_dirs': ['/usr/local/include'], 'library_dirs': ['/usr/local/lib'], 'runtime_library_dirs': [], 'extra_link_args': ['-Wl,-rpath', '-Wl,/usr/local/lib']}
    Configure: Autodetecting ZMQ settings...
        Custom ZMQ dir:       /usr/local
    clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -I/usr/local/include -Izmq/utils -c build/temp.macosx-14-arm64-cpython-312/scratch/vers.c -o build/temp.macosx-14-arm64-cpython-312/scratch/vers.o
    build/temp.macosx-14-arm64-cpython-312/scratch/vers.c:4:10: fatal error: 'zmq.h' file not found
    #include "zmq.h"
             ^~~~~~~
    1 error generated.
  
    error: command '/opt/homebrew/Library/Homebrew/shims/mac/super/clang' failed with exit code 1
  
    Warning: Couldn't find an acceptable libzmq on the system.
  
    If you expected pyzmq to link against an installed libzmq, please check to make sure:
  
        * You have a C compiler installed
        * A development version of Python is installed (including headers)
        * A development version of ZMQ >= 3.2 is installed (including headers)
        * If ZMQ is not in a default location, specify the env  ZMQ_PREFIX=<path>
        * If you did recently install ZMQ to a default location,
          try rebuilding the ld cache with `sudo ldconfig`
          or specify zmq's location with `ZMQ_PREFIX=/usr/local`
  
    You can skip all this detection/waiting nonsense if you know
    you want pyzmq to bundle libzmq as an extension by passing:
  
        `ZMQ_PREFIX=bundled`
  
    I will now try to build libzmq as a Python extension
    unless you interrupt me (^C) in the next 10 seconds...
  
    ************************************************
    ************************************************
    ************************************************
  
    10...
     9...
     8...
     7...
     6...
     5...
     4...
     3...
     2...
     1...
    Using bundled libzmq
    already have bundled/zeromq
    attempting ./configure to generate platform.hpp
    Warning: failed to configure libzmq:
    [Errno 2] No such file or directory: './configure'
    staging platform.hpp from: /private/tmp/locust--pyzmq-202[3112](https://github.com/Homebrew/homebrew-core/actions/runs/7004726049/job/19053226567#step:3:3113)7-6626-4putkf/pyzmq-25.1.1/buildutils/include_darwin
    ERROR setuptools_scm._file_finders.git listing git files failed - pretending there aren't any
    reading manifest file 'pyzmq.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no previously-included files found matching 'bundled/zeromq/src/Makefile*'
    adding license file 'LICENSE.BSD'
    adding license file 'LICENSE.LESSER'
    adding license file 'AUTHORS.md'
    writing manifest file 'pyzmq.egg-info/SOURCES.txt'
    ************************************************
```